pandas/core/frame.py

```diff
@@ -4678,22 +4678,16 @@ def drop_duplicates(
         inplace = validate_bool_kwarg(inplace, "inplace")
         duplicated = self.duplicated(subset, keep=keep)
 
-        if inplace:
-            (inds,) = np.asarray(-duplicated).nonzero()
-            new_data = self._data.take(inds)
+        result = self[-duplicated]
+        if ignore_index:
+            result.index = range(len(result))
```

Comment:
Are we moving away from `ibase. default_index`?

Reply:
good point, i should stick with the existing pattern, will update